src/Microsoft.AspNet.Mvc.Core/DefaultControllerActivator.cs

```diff
@@ -3,106 +3,26 @@
 
 using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
-using System.Reflection;
-using Microsoft.AspNet.Http;
-using Microsoft.AspNet.Mvc.Core;
-using Microsoft.AspNet.Mvc.ModelBinding;
 using Microsoft.Framework.DependencyInjection;
 
 namespace Microsoft.AspNet.Mvc
 {
     /// <summary>
-    /// Represents the <see cref="IControllerActivator"/> that is registered by default.
+    /// <see cref="IControllerActivator"/> that uses type activation to create controllers.
     /// </summary>
     public class DefaultControllerActivator : IControllerActivator
     {
-        private readonly Func<Type, PropertyActivator<ActionContext>[]> _getPropertiesToActivate;
-        private readonly IDictionary<Type, Func<ActionContext, object>> _valueAccessorLookup;
-        private readonly ConcurrentDictionary<Type, PropertyActivator<ActionContext>[]> _injectActions;
+        private static readonly Func<Type, ObjectFactory> _createControllerFactory =
```

Comment:
Up to you guys. My bet is that you can't even see the allocation anywhere on the trace (as this happens once per request), and there is no real need to cache. But oh well, it's fine.